packages/doenetml-worker-rust/lib-doenetml-core/src/core/state/types/math_expr.rs

```diff
@@ -0,0 +1,429 @@
+use std::collections::HashMap;
+use strum_macros::Display;
+
+use crate::math_via_wasm::{
+    eval_js, math_to_latex, normalize_math, parse_latex_into_math, parse_text_into_math,
+    substitute_into_math,
+};
+
+const BLANK_MATH_OBJECT: &str = "{\"objectType\":\"math-expression\",\"tree\":\"\u{ff3f}\"}";
+/// A mathematical expression represented as a serialized `math-expressions` object.
```

Comment:
Blank line above.

This seems like the most important item in this module. Could the docstring reflect that?

Like: `A mathematical symbolic mathematical expression allowing variables etc.. Operations on this object are carried out by the `math-expressions` JavaScript library, which allows for symbolic calculations like simplification. ... Data in this object is stored as a serialized string which is deserialized to preform operations. This may have performance implications and may be replaced with a rust-based symbolic calculator in the future`